File: backend/app/models/publication.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class PublicationBase(BaseModel):
    content: Optional[str]
    image: Optional[str]



class Publication(PublicationBase):
    id: int
    user_id: int
    time_posted: datetime
    class Config:
        orm_mode = True


class PublicationCreate(PublicationBase):
    pass


class PublicationLikes(BaseModel):
    user_ids: List[int]

File: frontend-react/src/interfaces/publication.ts
export interface IPublicationResponse {
    content: string | null;
    image: string | null;
    id: number;
    user_id: number;
    time_posted: string;
}

export interface IPublication {
    content: string | null;
    image: string | null;
    id: number;
    userId: number;
    timePosted: Date;
}
